util/release-odemis

```diff
@@ -0,0 +1,130 @@
+# Note: for a new major release (eg v2.7), it does NOT create a new branch
+# Create it (after updating the changelog) with:
+# git checkout -b v2.7-branch $commit-number-to-branch-from$
+#
+# It will:
+#  1. Tag the git repo with the changelog as description
+#  2. Push to the upstream remote
+#  3. Switch to ../pkg-native/odemis/ and update the repo
+#  4. Build for bionic (18.04), focal (20.04) and jammy (22.04).
+
+if [ ! -d ~/development/pkg-native/odemis ]; then
+     echo "Need directory ~/development/pkg-native/odemis with a clone of the odemis repository. Creating one now..."
+     # Do so with:
+     mkdir -p ~/development/pkg-native/
+     cd ~/development/pkg-native/
+     git clone https://github.com/delmic/odemis.git
+fi
+
+cd ~/development/odemis
+
+# Uncomment this line to run line-per-line
+# trap 'read -p "run: $BASH_COMMAND"' DEBUG
+
+# Get the version number from the debian/changelog file
+PVERSION=$(dpkg-parsechangelog -S Version) # should be like "3.1.2-1"
+if ! [[ "$PVERSION" =~ .*-1$ ]]; then
+    echo "Version should finish by -1, but found $PVERSION"
+    exit 1
+fi
+
+# Drop the -1
+RELVER=${PVERSION:0:-2} # Should be like "3.1.2"
+if ! [[ "$RELVER" =~ [2-9]\.[0-9]+\.[0-9]+$ ]] ; then
+    echo "Version $RELVER doesn't seem proper" 
+    exit 1
+fi
+
+# Check everything is commited
+if [ -n "$(git diff)" ]; then
+    echo "Tree no clean, make sure to commit all the changes first"
+    exit 1
+fi
+
+# check the branch is correct
+VBRANCH=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
+if ! [[ "$VBRANCH" =~ v.*-branch$ ]]; then
+    echo "Branch must be named vN.M-branch."
+    exit 1
+fi
+
+# check that a the version in setup.py is correct
+if ! grep -q "VERSION = \"$RELVER\"$" setup.py; then
+    echo "setup.py should contain the version $RELVER"
+    exit 1
+fi
+
+# Use the change log to annotate the tag
+echo "==================================================================="
+dpkg-parsechangelog -S Changes | tail -n +4
+echo "Will use the above changelog for tag v$RELVER: [Yn] "
+read answer
+if [[ $answer =~ ^[nN].* ]]; then
+    exit 2
+fi
+dpkg-parsechangelog -S Changes | tail -n +4 | git tag -a v$RELVER -F -
+
+# TODO: for a minor release, this is sufficient:
+#git push upstream
+git push --set-upstream upstream $VBRANCH
+git push --tags upstream
+
+# Go to package directory, and build the debian package
+cd ~/development/pkg-native/odemis
+
+git checkout $VBRANCH
+git pull
+
+BPVERSION=$(dpkg-parsechangelog -S Version) # should start with "$RELVER-1"
+if [[ "$BPVERSION" != "$PVERSION" ]]; then
+    echo "Failed to get version $PVERSION on the build directory"
+    exit 1
+fi
+
+read -p "Build odemis $RELVER and upload it? [Yn] " answer
+if [[ $answer =~ ^[nN].* ]]; then
+    exit 2
+fi
+git archive --prefix=odemis/ -o ../odemis_${RELVER}.orig.tar.gz HEAD || exit 1
+debuild -S || exit 1
+
+# TODO: if the version ends by ".0", push to proposed
+dput ppa:delmic-soft/odemis ../odemis_${PVERSION}_source.changes
```

Comment:
Just a comment, no need to address it. In future if we drop `bionic` then the default build will be for `focal` and so on. Maybe its better to get the `Series` from the (changelog)[https://github.com/delmic/odemis/blob/f2c0a35f5e5fd8a62a3d4c45396db9987c691d9e/debian/changelog#L1] and then build for next Ubuntu versions?

Reply:
Indeed. For now, I just manually update it. It takes about 5 min every 2 years... so no rush to automate that ;-)